examples/spectrogram_graph.cu

```diff
@@ -55,116 +55,160 @@ int main([[maybe_unused]] int argc, [[maybe_unused]] char **argv)
-  index_t nfft = 256;
-  index_t noverlap = nperseg / 8;
-  index_t nstep = nperseg - noverlap;
-  constexpr uint32_t num_iterations = 20;
-  float time_ms;
-
-  std::array<index_t, 1> num_samps{N};
-  std::array<index_t, 1> half_win{nfft / 2 + 1};
-  std::array<index_t, 1> s_time_shape{(N - noverlap) / nstep};
-
-  tensor_t<float, 1> time({N});
-  tensor_t<float, 1> modulation({N});
-  tensor_t<float, 1> carrier({N});
-  tensor_t<float, 1> noise({N});
-  tensor_t<float, 1> x({N});
-  auto freqs = make_tensor<float>(half_win);
-  tensor_t<complex, 2> fftStackedMatrix(
-      {(N - noverlap) / nstep, nfft / 2 + 1});
-  tensor_t<float, 1> s_time({(N - noverlap) / nstep});
-
-  // Set up all static buffers
-  // time = np.arange(N) / float(fs)
-  (time = linspace<0>(num_samps, 0.0f, static_cast<float>(N) - 1.0f) / fs)
-      .run(stream);
-  // mod = 500 * np.cos(2*np.pi*0.25*time)
-  (modulation = 500 * cos(2 * M_PI * 0.25 * time)).run(stream);
-  // carrier = amp * np.sin(2*np.pi*3e3*time + modulation)
-  (carrier = amp * sin(2 * M_PI * 3000 * time + modulation)).run(stream);
-  // noise = 0.01 * fs / 2 * np.random.randn(time.shape)
-  (noise = sqrt(0.01 * fs / 2) * random<float>({N}, NORMAL)).run(stream);
-  // noise *= np.exp(-time/5)
-  (noise = noise * exp(-1.0f * time / 5.0f)).run(stream);
-  // x = carrier + noise
-  (x = carrier + noise).run(stream);
-
-  for (uint32_t i = 0; i < 2; i++) {
-    // Record graph on second loop to get rid of plan caching in the graph
-    if (i == 1) {
-      cudaStreamBeginCapture(stream, cudaStreamCaptureModeGlobal);
-    }
-
-    // DFT Sample Frequencies (rfftfreq)
-    (freqs = (1.0 / (static_cast<float>(nfft) * 1 / fs)) *
-               linspace<0>(half_win, 0.0f, static_cast<float>(nfft) / 2.0f))
-        .run(stream);
-
-    // Create overlapping matrix of segments.
-    auto stackedMatrix = x.OverlapView({nperseg}, {nstep});
-    // FFT along rows
-    fft(fftStackedMatrix, stackedMatrix, 0, stream);
-    // Absolute value
-    (fftStackedMatrix = conj(fftStackedMatrix) * fftStackedMatrix)
-        .run(stream);
-    // Get real part and transpose
-    auto Sxx = fftStackedMatrix.RealView().Permute({1, 0});
-
-    // Spectral time axis
-    (s_time = linspace<0>(s_time_shape, static_cast<float>(nperseg) / 2.0f,
-                           static_cast<float>(N - nperseg) / 2.0f + 1) /
-                fs)
-        .run(stream);
-
-    if (i == 1) {
-      cudaStreamEndCapture(stream, &graph);
-      cudaGraphInstantiate(&instance, graph, NULL, NULL, 0);
-
-#if MATX_ENABLE_VIZ
-      // Generate a spectrogram visualization using a contour plot
-      viz::contour(time, freqs, Sxx);
-#else
-      printf("Not outputting plot since visualizations disabled\n");
-#endif            
-    }
-  }
-
-  auto f = make_tensor<float>({2000,2000});
-  auto f2 = make_tensor<float>({2000,2000});
-
-  (f = random<float>({2000,2000}, NORMAL)).run(stream);
-  inv(f2, f, stream);
+//   cudaEvent_t start, stop;
```

Comment:
Why is all those commented code left?  If it is stale let's delete it.

Reply:
I'm going to revert this file almost entirely once done